core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -334,6 +335,9 @@ private TableProperties() {}
   public static final String MAX_REF_AGE_MS = "history.expire.max-ref-age-ms";
   public static final long MAX_REF_AGE_MS_DEFAULT = Long.MAX_VALUE;
 
+  public static final String DELETE_GRANULARITY = "write.delete.granularity";
```

Comment:
I am still debating the property name. As it stands today, it will be applicable only to position deletes but I am not sure the name has to reflect it.

Reply:
Also, we would probably want to always use the file granularity for Flink position deletes to solve compaction issues.

Reply:
Maybe just `write.position-delete.granularity`? I prefer to use a more precise name and limit the scope of its usage.

A while ago I encountered an issue about adjusting the row-group size of Parquet position delete files.
I want to adjust the default row-group size of Parquet pos delete of the tables that I manage to speed up queries (more details are in issue #9149), however I found the parameter `write.delete.parquet.row-group-size-bytes` that controls the row-group size of Parquet pos delete also controls the row-group size of equality delete files. But obviously the row-group sizes applicable to these two type of delete files are not the same. 

Because we also use equality delete when the data size is small, I cannot directly set a default value of `write.delete.parquet.row-group-size-bytes` for new tables. I can only adjust `write.delete.parquet.row-group-size-bytes` according to the specific use of each table, which is inconvenient.

In fact, I think it is not appropriate to use one parameter to control the row-group size of both position delete files and equality delete files, so I created  #9177 to add a separate parameter for the position delete file that only writes the `file_path` and `pos` columns.

Back to this, IIUC, If we later add a grouping granularity for equality delete, since position delete and equality delete have different characteristics, they will most likely apply different grouping granularity. So I think we'd better make the distinction right from the start, what do you think?

Reply:
To be honest, I doubt we will ever support this property for equality deletes.

In general, I do get that we may want to configure position and equality deletes differently. We can explore adding an extra namespace. I am still not sure this use case falls into that bucket.

@rdblue @RussellSpitzer @zhongyujiang, thoughts? Do we want a prefix for this config to make it explicit that it only applies to position deletes? Currently, I only note that in the docs.

Reply:
This option makes no sense for equality deletes because they aren't targeted at a single file, so I agree that we won't support it for equality. This is also mostly advisory. It is unlikely that we will support it in Flink and will instead always use file-level granularity. Maybe we won't even want to support this in the long term, if we decide that Spark performs better with file granularity at all times.

I guess where I'm at for this is that I would probably not worry much about it -- but also not add it to documentation since we will probably not want people setting it themselves. I think I'd leave it as `write.delete.granularity`.

Reply:
The idea not to document it for now is reasonable given that it acts like a recommendation and we are not sure we want to support it in the future. Let's keep the name as is then.

Adding a way to configure position and equality deletes separately is another discussion.